### ISSUE TITLE
[BUG] The name property changes depending on project folder name in package-lock.json

### ISSUE THREAD

**Issue by csvn** [npm/arborist#311]
# What / Why

When working on a project it is sometimes nice to clone an extra version of the project with Git (for reviewing code, etc), and thus naming it differently from usual. Running `npm install` in both project folders (where `package.json` is located) produces different lockfile results; even when there are no changes. If the name property in `package.json` and the project folder's name are equal, it will leave out the `packages."".name` property. This makes it feel like the installation is unreliable, as simply running an install may produce unstaged changes to the lockfile that needs to be reverted.

## When

* n/a

## Where

* n/a

### Current Behavior

If folder name is the same as the project name, the `package-lock.json` file looks like this:
```json
  "packages": {
    "": {
      "version": "0.0.0"
    }
  }
```
If not, the package name will be added to `packages.""` in `package-lock.json`:
```json
  "packages": {
    "": {
      "name": "foo",
      "version": "0.0.0"
    }
```

### Steps to Reproduce

1. Create 2 folders; e.g. `foo` and `foo2` (to simulate cloning multiple projects)
2. Add the following `package.json` file to both folders:
   ```json
   {
     "name": "foo",
     "version": "0.0.0"
   }
   ```
3. Run `npm install` in both folders
4. Now the `foo` folder is missing `package."".name`, while the `foo2` folder has the property included

### Expected Behavior

No matter the name of the project folder where `package.json` is located, the `package-lock.json` looks like:

```json
  "packages": {
    "": {
      "version": "0.0.0"
    }
  }
```

## Who

* n/a

## References

* n/a

**Comment by csvn** [comment]
I think [these lines](https://github.com/npm/arborist/blob/80f5029fc02274e5c1068891ee16ec2f42c4a534/lib/shrinkwrap.js#L262-L263) are the relevant ones here, but I'm avoiding making a PR for this since I'm unsure if changing this may have side-effects for other code.

**Comment by ljharb** [comment]
This has recently been fixed, i believe - which version of npm are you testing with?

**Comment by csvn** [comment]
@ljharb I was on `7.20.3`. I just tried installing `latest` (`7.21.0`), and then running `npm install` in both folders. It seems that the name property is now always included (below).

```json
  "packages": {
    "": {
      "name": "foo",
      "version": "0.0.0"
    }
  }
```

I'm surprised that this was already fixed, I thought I updated very recently. Thanks for pointing that out!

**Comment by csvn** [comment]
It was probably [fixed in `7.20.6`](https://github.com/npm/cli/releases/tag/v7.20.6), so that was only a week ago. Funny timing 😅

> fix(shrinkwrap): always set name on the root node
